test/e2e/etcd_backup_test.go

```diff
@@ -49,32 +48,11 @@ var _ = Describe("Etcd Backup", func() {
 			provider := p
 			Context(fmt.Sprintf("with provider %s", provider.Name), func() {
 				BeforeEach(func() {
-					etcdName = fmt.Sprintf("etcd-%s", provider.Name)
+					etcdName = fmt.Sprintf("etcd-single-node-%s", provider.Name)
+					storePrefix = etcdName
 					storageContainer = getEnvAndExpectNoError(envStorageContainer)
 
-					By("Purge snapstore")
```

Comment:
You have moved this piece of code to `purgeSnapshotIfNeeded`.  The said function is unfortunately defined in `etcd_multi_node_test.go` which is a bit weird. Why should one test have dependency on another? I can see that such dependencies already exist, then you have just added one more reason for that dependency to exist :(

Is this change required to make tests run concurrently? If no, then please revert back this change.

Reply:
In [test/e2e/etcd_backup_test.go](./test/e2e/etcd_backup_test.go), it doesn’t make sense to include the `purgeSnapshotIfNeeded` function since the file contains only one test case. However, if you look at [test/e2e/etcd_multi_node_test.go](./test/e2e/etcd_multi_node_test.go), it houses multiple tests that utilize the same `BeforeEach` setup, with each test requiring a distinct etcd name. This is why I moved the function inside the test case and refactored it into a simpler form.

I acknowledge that the function is defined in [./test/e2e/etcd_backup_test.go](./test/e2e/etcd_backup_test.go), where `purgeSnapshotIfNeeded` and its dependent functions are located. To avoid code duplication and utilize the function efficiently across every occurrence, I decided not to repeat the code but rather to define it where it can be used by multiple test cases.

Reply:
Ok generally if you have a test utility function that is used across test files then you ideally define it in utils package/file. For now i will let this go and I will perhaps make that change during overhauling the e2e tests.